app/services/katello/candlepin/message_handler.rb

```diff
@@ -73,7 +77,13 @@ def import_pool(index_hosts = true)
         if pool
           ::Katello::EventQueue.push_event(::Katello::Events::ImportPool::EVENT_TYPE, pool.id)
         else
-          ::Katello::Pool.import_pool(pool_id, index_hosts)
+          begin
+            ::Katello::Pool.import_pool(pool_id, index_hosts)
+          rescue ActiveRecord::RecordInvalid
+            # if we hit this block it's likely that the pool's subscription, product are being created
+            # as a result of manifest import/refresh or custom product creation
+            Rails.logger.warn("Unable to import pool. It will likely be created by another process.")
```

Comment:
What would happen if it's not created by another process?

Reply:
Then that is an error and will be raised accordingly by product creation, manifest import, and so on. I can't think of a way that those actions would even raise that error because everything is happening synchronously - so it signifies a serious problem. At this point in the code there's nothing we can do but log it (and avoid unnecessary stack trace)